main/src/main/java/cgeo/geocaching/connector/gc/GCParser.java

```diff
@@ -1703,6 +1727,34 @@ private static void mergeFriendsLogs(final List<LogEntry> mergedLogs, final Iter
         }
     }
 
+    private static void mergeLogTimes(final List<LogEntry> mergedLogTimes, final Iterable<LogEntry> logTimesToMerge) {
+        for (final LogEntry logToMerge : logTimesToMerge) {
```

Comment:
Quadratic complexity.

Maybe use a lookup hashtable by logId to prevent this?

Reply:
Good idea, will  check on that

Reply:
Done